### ISSUE TITLE
JSONDecodeError when running autsklearn regressor

### ISSUE THREAD

**Issue by yakdahl** [automl/auto-sklearn#712]
Hi all, 

I really like this code! I've been running it for a couple of scientific problems, and am using it for chain regression in a multidimensional dataset while using a custom scoring function. It works fine when I run for a short time (time_left_for_this_task = ~360 s or less), but when I increase much past that the code will occasionally crash with the following error message: 

JSONDecodeError: Expecting ':' delimiter: line 1 column 197051 (char 197050) 

(I don't think the characters are always the same, I don't have another error open to compare right now. It does happen frequently with longer run times, and I believe more frequently the longer I run it)
Any ideas about what might be causing this?

**Comment by yakdahl** [comment]
Sorry, I realize I did not print the full error message:


JSONDecodeError                           Traceback (most recent call last)
<ipython-input-52-de0460cf24d8> in <module>
      8     ASKLR = autosklearn.regression.AutoSklearnRegressor(time_left_for_this_task = 600, per_run_time_limit = 90,
      9                                                         n_jobs = 20)
---> 10     ASKLR.fit(X_trainx,y_train[i])
     11     ASKLR.refit(X_trainx,y_train[i])
     12     pipelines6.append([i,copy.deepcopy(ASKLR),X_trainx,X_testx,y_train[i],y_test[i]])

~/.local/lib/python3.7/site-packages/autosklearn/estimators.py in fit(self, X, y, X_test, y_test, metric, feat_type, dataset_name)
    800             metric=metric,
    801             feat_type=feat_type,
--> 802             dataset_name=dataset_name,
    803         )
    804 

~/.local/lib/python3.7/site-packages/autosklearn/estimators.py in fit(self, **kwargs)
    397                 automl=self._automl[0],
    398                 kwargs=kwargs,
--> 399                 load_models=True,
    400             )
    401             for p in processes:

~/.local/lib/python3.7/site-packages/autosklearn/estimators.py in _fit_automl(automl, kwargs, load_models)
     13 
     14 def _fit_automl(automl, kwargs, load_models):
---> 15     return automl.fit(load_models=load_models, **kwargs)
     16 
     17 

~/.local/lib/python3.7/site-packages/autosklearn/automl.py in fit(self, X, y, X_test, y_test, metric, feat_type, dataset_name, only_return_configuration_space, load_models)
   1089             dataset_name=dataset_name,
   1090             only_return_configuration_space=only_return_configuration_space,
-> 1091             load_models=load_models,
   1092         )
   1093 

~/.local/lib/python3.7/site-packages/autosklearn/automl.py in fit(self, X, y, task, metric, X_test, y_test, feat_type, dataset_name, only_return_configuration_space, load_models)
    206             metric=metric,
    207             load_models=load_models,
--> 208             only_return_configuration_space=only_return_configuration_space,
    209         )
    210 

~/.local/lib/python3.7/site-packages/autosklearn/automl.py in _fit(self, datamanager, metric, load_models, only_return_configuration_space)
    487             )
    488             self.runhistory_, self.trajectory_ = \
--> 489                 _proc_smac.run_smbo()
    490             trajectory_filename = os.path.join(
    491                 self._backend.get_smac_output_directory_for_run(self._seed),

~/.local/lib/python3.7/site-packages/autosklearn/smbo.py in run_smbo(self)
    511                 output_dirs=smac.solver.scenario.input_psmac_dirs,
    512                 configuration_space=smac.solver.config_space,
--> 513                 logger=smac.solver.logger,
    514             )
    515 

~/.local/lib/python3.7/site-packages/smac/optimizer/pSMAC.py in read(run_history, output_dirs, configuration_space, logger)
     49                                                file_in_output_directory)
     50                 run_history.update_from_json(runhistory_file,
---> 51                                              configuration_space)
     52 
     53                 new_numruns_in_runhistory = len(run_history.data)

~/.local/lib/python3.7/site-packages/smac/runhistory/runhistory.py in update_from_json(self, fn, cs, origin)
    379         """
    380         new_runhistory = RunHistory(self.aggregate_func)
--> 381         new_runhistory.load_json(fn, cs)
    382         self.update(runhistory=new_runhistory, origin=origin)
    383 

~/.local/lib/python3.7/site-packages/smac/runhistory/runhistory.py in load_json(self, fn, cs)
    346         """
    347         with open(fn) as fp:
--> 348             all_data = json.load(fp, object_hook=StatusType.enum_hook)
    349 
    350         self.ids_config = {int(id_): Configuration(cs, values=values)

/home/mgcf/software-ws/anaconda/anaconda3/lib/python3.7/json/__init__.py in load(fp, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)
    294         cls=cls, object_hook=object_hook,
    295         parse_float=parse_float, parse_int=parse_int,
--> 296         parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
    297 
    298 

/home/mgcf/software-ws/anaconda/anaconda3/lib/python3.7/json/__init__.py in loads(s, encoding, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)
    359     if parse_constant is not None:
    360         kw['parse_constant'] = parse_constant
--> 361     return cls(**kw).decode(s)

/home/mgcf/software-ws/anaconda/anaconda3/lib/python3.7/json/decoder.py in decode(self, s, _w)
    335 
    336         """
--> 337         obj, end = self.raw_decode(s, idx=_w(s, 0).end())
    338         end = _w(s, end).end()
    339         if end != len(s):

/home/mgcf/software-ws/anaconda/anaconda3/lib/python3.7/json/decoder.py in raw_decode(self, s, idx)
    351         """
    352         try:
--> 353             obj, end = self.scan_once(s, idx)
    354         except StopIteration as err:
    355             raise JSONDecodeError("Expecting value", s, err.value) from None

JSONDecodeError: Expecting ':' delimiter: line 1 column 69928 (char 69927)

**Comment by panxhong** [comment]
same problem

**Comment by mfeurer** [comment]
This is a bug in SMAC. You can add a try/except around
```
~/.local/lib/python3.7/site-packages/autosklearn/smbo.py in run_smbo(self)
511 output_dirs=smac.solver.scenario.input_psmac_dirs,
512 configuration_space=smac.solver.config_space,
--> 513 logger=smac.solver.logger,
514 )
515
```
to temporarily prevent this. We'll add a fix in a new SMAC version.

**Comment by mfeurer** [comment]
We fixed this in the latest release (0.7.0) by using the latest SMAC version. Please reopen if this issue persists with auto-sklearn 0.7.0.
